src/shogun/mathematics/linalg/LinalgBackendBase.h

```diff
@@ -49,58 +49,48 @@ namespace shogun
 class LinalgBackendBase
 {
 public:
+	#define DEFINE_FOR_ALL_PTYPE(METHODNAME, Container) \
+	METHODNAME(bool, Container); \
+	METHODNAME(char, Container); \
+	METHODNAME(int8_t, Container); \
+	METHODNAME(uint8_t, Container); \
+	METHODNAME(int16_t, Container); \
+	METHODNAME(uint16_t, Container); \
+	METHODNAME(int32_t, Container); \
+	METHODNAME(uint32_t, Container); \
+	METHODNAME(int64_t, Container); \
+	METHODNAME(uint64_t, Container); \
+	METHODNAME(float32_t, Container); \
+	METHODNAME(float64_t, Container); \
+	METHODNAME(floatmax_t, Container); \
+	METHODNAME(complex128_t, Container); \
```

Comment:
I currently defined the `DEFINE_FOR_ALL_PTYPE` 3 times separately in LinalgBackendBase, Eigen and ViennaCL, because it is easy to control (undefine) and many types don't work for ViennaCL.